assignments/02-collections-lifetimes/src/life.rs

```diff
@@ -1,19 +1,83 @@
-fn split_string(string: &str, delimeter: &str) -> Vec<&str> {
-    todo!()
+fn split_string<'a>(string: &'a str, delimiter: &str) -> Vec<&'a str> {
+    string.split(delimiter).filter(|s| !s.is_empty()).collect()
 }
 
 #[derive(PartialEq, Debug)]
-struct Differences {
-    only_in_first: Vec<&str>,
-    only_in_second: Vec<&str>,
+struct Differences<'a> {
+    only_in_first: Vec<&'a str>,
+    only_in_second: Vec<&'a str>,
 }
```

Comment:
note - this works fine, but since these two don't technically need to live in the same lifetime we can also separate them into two lifetimes as so:

```
struct Differences<'a, 'b> {
    only_in_first: Vec<&'a str>,
    only_in_second: Vec<&'b str>,
}
```

(not counting off, just wanted to point this out bc it could be useful for future functions)